tools/mksurfdata_esmf/src/mksoiltexMod.F90

```diff
@@ -200,7 +205,8 @@ subroutine mksoiltex(file_mesh_i, file_mapunit_i, file_lookup_i, mesh_o, pioid_o
 
     ! Determine mapunit_value_max (set it as a module variable so that it can be
     ! accessible to gen_dominant_mapunit) - this is needed in the dynamic mask routine
-    mapunit_value_max = maxval(dataptr)
+    mapunit_value_max_local = maxval(dataptr)
+    call mpi_allreduce(mapunit_value_max_local, mapunit_value_max, 1, MPI_INTEGER, MPI_MAX, mpicom, rcode)
```

Comment:
This is the key change that fixes https://github.com/ESCOMP/CTSM/issues/2744. @mvertens and I aren't sure why the impact is substantial for SE grids and not for FV grids, but it's an important fix for all mappings. The issue is that the processor-local values in dataptr don't encompass the entire range of possible mapunits, and in some cases are all 0s... the all 0s case shows up in a very clear way, whereas other cases can show up as more subtle issues.

Reply:
Thanks for explaining this in your PR.

Reply:
@billsacks it looks to me like a little comment just before the mpi_allreduce about why it's needed would be good here. For robustness, checking rcode would also be best practice.

I can do that if you aren't able to get to it.